api/app/events/api.py

```diff
@@ -96,13 +99,86 @@ def get(self):
         return returnEvents, 200
 
 
+class EventAPI(EventMixin, restful.Resource):
+    @auth_required
+    def post(self):
+        args = self.req_parser.parse_args()
+
+        user_id = g.current_user["id"]
+        current_user = user_repository.get_by_id(user_id)
+        if not current_user.is_admin:
+            return FORBIDDEN
+
+        name = args['name']
+        description = args['description']
+        start_date = datetime.strptime(
+            (args['start_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        end_date = datetime.strptime(
+            (args['end_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+
+        event = Event(name=name,
+                      description=description,
+                      start_date=start_date,
+                      end_date=end_date)
+
+        db.session.add(event)
+        db.session.commit()
```

Comment:
Can we add a new method called `add(event: Event)` to the ./api/event/repository? And then import the method in. Personally I prefer to push lower level operations down and away from an API level. I think it looks it makes the logic clearer and allows re-usability of the methods. What do you think?

Reply:
Sounds good. I like this design pattern. I see that this might need to be applied to the other APIs, such as users.